java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -1157,62 +1157,61 @@ private void handleImageBuildData(ServerAction serverAction, JsonElement jsonRes
+                return;
+            }
+            Optional<Long> imageProfileId = Optional.ofNullable(details.get().getImageProfileId());
+            if (imageProfileId.isEmpty()) { // It happens when the image profile is deleted during a build action
+                LOG.warn("Image Profile ID not found while performing: "  +
+                        action.getName() + " in handleImageBuildData");
+                return;
+            }
+            Optional<ImageProfile> profileOpt =
+                    ImageProfileFactory.lookupById(imageProfileId.get());
+            profileOpt.ifPresent(p -> p.asKiwiProfile().ifPresent(kiwiProfile -> {
+                serverAction.getServer().asMinionServer().ifPresent(minionServer -> {
+                    // Download the built Kiwi image to SUSE Manager server
+                    OSImageInspectSlsResult.Bundle bundleInfo =
+                            Json.GSON.fromJson(jsonResult, OSImageBuildSlsResult.class)
+                                    .getKiwiBuildInfo().getChanges().getRet().getBundle();
+                    infoOpt.ifPresent(info -> info.setChecksum(
+                            ImageInfoFactory.convertChecksum(bundleInfo.getChecksum())));
+                    MgrUtilRunner.ExecResult collectResult = systemQuery
+                            .collectKiwiImage(minionServer, bundleInfo.getFilepath(),
+                                    OSImageStoreUtils.getOsImageStorePath() + kiwiProfile.getTargetStore().getUri())
+                            .orElseThrow(() -> new RuntimeException("Failed to download image."));
+                    if (collectResult.getReturnCode() != 0) {
+                        serverAction.setStatus(ActionFactory.STATUS_FAILED);
+                        serverAction.setResultMsg(StringUtils
+                                .left(printStdMessages(collectResult.getStderr(), collectResult.getStdout()),
+                                        1024));
+                    }
                 });
-            }, () -> {
-                LOG.error("Details not found in ImageBuildAction");
-                LOG.error("Name is: " + action.getName());
-                LOG.error("Action ID is: " + action.getId());
-                LOG.error("Earliest action was: " + action.getEarliestAction());
-                LOG.error("Scheduler User is: " + action.getSchedulerUser());
+            }));
+            ImageInspectAction iAction = ActionManager.scheduleImageInspect(
+                    action.getSchedulerUser(),
+                    action.getServerActions()
+                            .stream()
+                            .map(ServerAction::getServerId)
+                            .collect(Collectors.toList()),
+                    Optional.of(action.getId()),
+                    details.get().getVersion(),
+                    profileOpt.map(ImageProfile::getLabel).orElse(null),
+                    profileOpt.map(ImageProfile::getTargetStore).orElse(null),
```

Comment:
This might generate another NPE because from what I see `ActionManager.scheduleImageInspect()` [cannot handle a null `store`](https://github.com/mbussolotto/uyuni/blob/25e111d48108ea2b8c6a25cd81b4dad9a9c29143/java/code/src/com/redhat/rhn/manager/action/ActionManager.java#L2331). 

BTW can `profileOpt` actually be empty if we already checked that we have a valid `imageProfileId`?

Reply:
I think `profileOpt` could not be empty.. but all this part was present before my changes so I m not confident with that. 
Anyway, I guess you are proposing to remove `Optional` wrapping from `profileOpt`, right? Or do you think it is safer a simple and dirty (but really effective) check for `store==null` at the beginning of `scheduleImageInspect` ?

Reply:
> I guess you are proposing to remove Optional wrapping from profileOpt, right? Or do you think it is safer a simple and dirty (but really effective) check for store==null at the beginning of scheduleImageInspect ?

My functional knowledge is limited here I fear :sweat_smile: Does it make sense to schedule an image inspect action if the store is null? If not, then we should just skip this part if the optional is empty (maybe with a warning). Otherwise, we should deal with the null inside the method and create the action correctly.

Reply:
You cannot schedule an image inspect without the store to be available.
If profile or store are gone, the action details have just informational character. It is just historical data, you cannot do anything with it.

Reply:
now it should look better